Sources/Starknet/Providers/StarknetProvider/StarknetProvider.swift

```diff
@@ -111,4 +111,12 @@ public class StarknetProvider: StarknetProviderProtocol {
 
         return result
     }
+
+    public func getEvents(from fromBlockId: StarknetBlockId, to toBlockId: StarknetBlockId, address: Felt, keys: [Felt], chunkSize: UInt64, continuationToken: String?) async throws -> GetEventsResponse {
```

Comment:
Maybe it would be better to pass  a struct with these params here? as there are 6 of them

Reply:
I thought about that, but out of those 6 params, 2 (fromBlockId and toBlockId) are defaulted to "lastest" by devnet and on top of that continuation token is optional, so is it really necessary?

Reply:
If I read the starknet specs right, all fields apart from chunkSize are optional, so we could make Filter public, and
give it a public initializer with all fields optional and w/ default values, eg.:
```swift
struct Filter: Encodable {
    let fromBlockId: StarknetBlockId?
    let toBlockId: StarknetBlockId?
    let address: Felt?
    let keys: [Felt]?
    let chunkSize: UInt64
    let continuationToken: String?

    init(fromBlockId: StarknetBlockId? = nil, toBlockId: StarknetBlockId? = nil, address: Felt? = nil, keys: [Felt]? = nil, chunkSize: UInt64 = 50, continuationToken: String? = nil) {
        self.fromBlockId = fromBlockId
        self.toBlockId = toBlockId
        self.address = address
        self.keys = keys
        self.chunkSize = chunkSize
        self.continuationToken = continuationToken
    }

...
}
```

And then it would be passed to the getEvents function.